src/library/scala/util/Using.scala

```diff
@@ -122,21 +136,37 @@ object Using {
     */
   def apply[R](resource: => R): Using[R] = new Using(resource)
 
+  private def preferentiallySuppress(primary: Throwable, secondary: Throwable): Throwable = {
+    def score(t: Throwable): Int = t match {
+      case _: VirtualMachineError                   => 4
+      case _: LinkageError                          => 3
+      case _: InterruptedException | _: ThreadDeath => 2
+      case _: ControlThrowable                      => 0
+      case e if !NonFatal(e)                        => 1 // in case this method gets out of sync with NonFatal
+      case _                                        => -1
```

Comment:
Is there any chance to make this ` 4,3,2,1,0,-1` to be named constants?

Reply:
does that improve clarity? They're only used here, and I would think it's clearer when you can see their values (since they're not quite in sorted order)

Reply:
(because `ControlThrowable` sorts below other fatals, it needs to be matched before `NonFatal`, or they'll end up with the same score/rank)